src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -661,103 +766,129 @@ void CameraAvSettingsUserLevelMgmtServer::HandleMPTZRelativeMove(HandlerContext
+                         mEndpointId);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
             return;
         }
         int16_t tiltDeltaValue = tiltDelta.Value();
         if (tiltDeltaValue > (mTiltMax - mTiltMin) || tiltDeltaValue < -(mTiltMax - mTiltMin))
         {
-            ChipLogError(Zcl, "TiltDelta value received is out of range.");
+            ChipLogError(Zcl, "CameraAVSettingsUserLevelMgmt[ep=%d]: TiltDelta value received is out of range.", mEndpointId);
             ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
             return;
         }
 
-        newTilt = static_cast<int16_t>(newTilt + tiltDeltaValue);
-        if (newTilt > mTiltMax)
+        // If we're here, then we'll also have an existing Tilt value in MPTZPosition
+        // Note that newTilt is always between -180 and 180; and the tiltDeltaValue is constrained to -360 to 360 by the
+        // code above, so the addition sum can never overflow int16_t.
+        //
+        int16_t newTiltValue = static_cast<int16_t>(mMptzPosition.tilt.Value() + tiltDeltaValue);
+        if (newTiltValue > mTiltMax)
         {
-            newTilt = mTiltMax;
+            newTiltValue = mTiltMax;
         }
-        if (newTilt < mTiltMin)
+        if (newTiltValue < mTiltMin)
         {
-            newTilt = mTiltMin;
+            newTiltValue = mTiltMin;
         }
 
+        newTilt.Emplace(newTiltValue);
+
         hasAtLeastOneValue = true;
     }
 
     if (zoomDelta.HasValue())
     {
         if (!HasFeature(Feature::kMechanicalZoom))
         {
-            ChipLogError(Zcl, "Mechanical Zoom not supported although zoomDelta value provided");
-            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            ChipLogError(Zcl,
+                         "CameraAVSettingsUserLevelMgmt[ep=%d]: Mechanical Zoom not supported although zoomDelta value provided",
+                         mEndpointId);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
             return;
         }
         int8_t zoomDeltaValue = zoomDelta.Value();
         if (zoomDeltaValue > (mZoomMax - 1) || zoomDeltaValue < -(mZoomMax - 1))
         {
-            ChipLogError(Zcl, "ZoomDelta value received is out of range.");
+            ChipLogError(Zcl, "CameraAVSettingsUserLevelMgmt[ep=%d]: ZoomDelta value received is out of range.", mEndpointId);
             ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
             return;
         }
 
-        newZoom = static_cast<int8_t>(newZoom + zoomDeltaValue);
-        if (newZoom > mZoomMax)
+        // If we're here, then we'll also have an existing Pan value in MPTZPosition. The zoom values are constrained such that
+        // we won't overflow newZoomValue
+        //
+        int16_t newZoomValue = static_cast<int16_t>(mMptzPosition.zoom.Value() + zoomDeltaValue);
```

Comment:
So this is relying on some pretty subtle promotion mechanics (e.g. that the unsigned value gets promoted to a signed one automatically, because the unsigned type is sufficiently narrow, etc).  Might clearer as:
```suggestion
        int newZoomValue = static_cast<int>(mMptzPosition.zoom.Value()) + zoomDeltaValue;
```